src/shogun/util/factory.h

```diff
@@ -333,5 +335,63 @@ namespace shogun
 	{
 		return std::make_shared<PipelineBuilder>();
 	}
+
+	class StringsFeatures;
+	class FeatureSubset;
+
+	template <
+	    typename TypeName, typename... Args,
+	    std::enable_if_t<
+	        traits::is_any_of_v<
+	            TypeName, Features, StringsFeatures, FeatureSubset,
+	            FeatureSubset, CSVFile, LibSVMFile, PipelineBuilder, Kernel>,
+	        TypeName>* = nullptr>
+	std::shared_ptr<TypeName> create(Args&&... args)
+	{
+		if constexpr (std::is_same_v<TypeName, Features>)
+		{
+			return features(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, StringsFeatures>)
```

Comment:
would it make sense to generative these if then else statement automatically somehow using the fact that we can one-to-one relate these factory names with the type?